frame/elections-phragmen/src/lib.rs

```diff
@@ -471,19 +507,46 @@ decl_module! {
 		/// Writes: O(do_phragmen)
 		/// # </weight>
 		#[weight = (2_000_000_000, DispatchClass::Operational)]
-		fn remove_member(origin, who: <T::Lookup as StaticLookup>::Source) -> DispatchResult {
+		fn remove_member(
+			origin,
+			who: <T::Lookup as StaticLookup>::Source,
+			has_replacement: bool,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let who = T::Lookup::lookup(who)?;
 
+			let will_have_replacement = <RunnersUp<T>>::decode_len().unwrap_or(0) > 0;
+			match (will_have_replacement, has_replacement) {
```

Comment:
This is a root call... is this 3 statement match needed?

End of the day some governance or Sudo is going want to "remove_member". Reporting the accurate behavior is important for weight calculation here, but I dont see this input as something malicious, else it would be voted down

Reply:
Would have made both my work and weighing easier but I think we still have to make sure even a root call is checked. I.e. The runtime should do some checks to make sure that the root call is correct.